File: storage-service/src/main/java/com/nixsolutions/storageservice/config/WebSecurityConfig.java
package com.nixsolutions.storageservice.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.web.server.SecurityWebFilterChain;
import com.nixsolutions.financial.security.config.HttpSecurityConfigurationHolder;

@Configuration
public class WebSecurityConfig
{
  @Bean
  public SecurityWebFilterChain webFilterChain(HttpSecurityConfigurationHolder securityConfigurationHolder)
  {
    return securityConfigurationHolder.getHttpSecurity()
        .csrf().disable()
        .authorizeExchange().anyExchange().permitAll()
        .and().build();
  }
}

File: user-service/src/main/kotlin/com/nixsolutions/userservice/config/WebSecurityConfig.kt
package com.nixsolutions.userservice.config

import com.nixsolutions.financial.security.config.HttpSecurityConfigurationHolder
import org.springframework.boot.actuate.autoconfigure.security.reactive.EndpointRequest
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.web.server.ServerHttpSecurity
import org.springframework.security.web.server.SecurityWebFilterChain

@Configuration
class WebSecurityConfig {

  @Bean
  fun springSecurityFilterChain(httpSecurityHolder: HttpSecurityConfigurationHolder): SecurityWebFilterChain {
    return httpSecurityHolder.httpSecurity.csrf().disable()
        .authorizeExchange()
        .pathMatchers("/api/**").hasAuthority("admin_permission")
        .and()
        .httpBasic().disable()
        .build()
  }
}

